src/core_inner/backend/pearl/group.rs

```diff
@@ -242,22 +242,18 @@ impl PearlGroup {
                     trace!("get data: {} from: {}", data, holder);
                     results.push(data);
                 }
-                Err(err) if err != BackendError::KeyNotFound => {
+                Err(err) if err != BackendError::KeyNotFound(key) => {
                     has_error = true;
                     debug!("get error: {}, from : {}", err, holder);
                 }
                 _ => debug!("key not found from: {}", holder),
             }
         }
-        if results.is_empty() {
-            if has_error {
-                debug!("cannot read from some pearls");
-                Err(Error::Failed("cannot read from some pearls".to_string()))
-            } else {
-                Err(Error::KeyNotFound)
-            }
+        if results.is_empty() && has_error {
```

Comment:
Why error and empty results branches put together?

Reply:
I preserve the same logic. Settings::choose data returns result value with only KeyNotFound error. I changed return type to option and did equal logic transormation